actions.js

```diff
@@ -442,8 +442,11 @@ ActionManager.prototype._isBatchEvent = function(msg) {
 
 ActionManager.prototype.onReceiveAction = function(msg) {
     if (this.isPreviousAction(msg)) return;
-    if (this.isUserAction(msg) && !SnapUndo.contains(msg)) {
-        return SnapUndo.record(msg);
+    if (this.isUserAction(msg)) {
+        if (!SnapUndo.contains(msg)) {
+            SnapUndo.record(msg);
+        }
+        return;
```

Comment:
where else `_applyEvent(msg);` gets applied to `userAction`?

Reply:
Only [here](https://github.com/NetsBlox/Snap--Build-Your-Own-Blocks/blob/c63a04a5358fdcd0ee5bb5c1c3ccd1a6246089f3/actions.js#L328) but that is only for queued actions (user actions are never queued).

Reply:
So user actions were never supposed to be applied right?

Reply:
Correct. That was actually the problem :)